content/doc/book/system-administration/security/configuring-content-security-policy.adoc

```diff
@@ -0,0 +1,169 @@
+---
+title: Configuring Content Security Policy
+layout: documentation
+---
+ifdef::backend-html5[]
+:toc: left
+endif::[]
+
+Jenkins 1.641 / Jenkins 1.625.3 introduce the `Content-Security-Policy` header to static files served by Jenkins (specifically `DirectoryBrowserSupport`).
+This header is set to a very restrictive default set of permissions to protect Jenkins users from malicious HTML/JS files in workspaces, `/userContent`, or archived artifacts.
+
+Unfortunately, several popular, useful plugins are affected by this and lose part of their functionality unless the default rules are relaxed.
+
+*Alternatives*
+
+TIP: Since Jenkins 2.200, it is possible to define a *Resource Root URL* in the Jenkins system configuration as an alternative to relaxing the Content Security Policy rules.
+See its inline help or details.
+
+
+== The Default Rule Set
+
+The default rule is set to:
+
+[source,javascript]
+----
+sandbox; default-src 'none'; img-src 'self'; style-src 'self';
+----
+
+This rule set results in the following:
+
+* No JavaScript allowed at all
+* No plugins (object/embed) allowed
+* No inline CSS, or CSS from other sites allowed
+* No images from other sites allowed
+* No frames allowed
+* No web fonts allowed
+* No XHR/AJAX allowed
+* etc.
+
+In detail:
+
+* `sandbox` limits a number of things of what the page can do, similar to the `sandbox` attribute set on iframes.
+For a full list of what is prohibited, see https://html.spec.whatwg.org/multipage/browsers.html#sandboxing-flag-set[this site].
+This attribute is not widely supported.
+* `default-src 'none'` prohibits loading scripts, URLs for AJAX/XHR/WebSockets/EventSources, fonts, plugin objects, media, and frames from anywhere (images and styles would also be prohibited, but
+are allowed by more specific rules described below).
+* `img-src 'self'` allows loading images from other files served by Jenkins.
+Inline image definitions are prohibited.
+* `style-src 'self'` allows loading style sheets from other files served by Jenkins.
+Inline style sheets are prohibited.
+
+See http://content-security-policy.com/[content-security-policy.com] for a reference on this header and its possible values.
+
+== Getting things working
+
+The most expedient approach is to use Jenkins 2.200+ and set up a second domain pointing to the same Jenkins instance (Jenkins URL:build.example.com; Resource Root URL: build-artifacts.example.com).
+This will result in resources being served from the resource root URL instead of the Jenkins URL.
+The advantage of this is that there are no cookies associated with this domain, and file paths are hopefully sufficiently non predictable that people won't be able to exfiltrate content.
```

Comment:
That this is placed here is really weird. We're in the section on the default rule set.